### PR TITLE
[1.10] Append wall entry controls to the file context menu

### PR DESCRIPTION
**Pull request by yurabakhtin**
Issue: https://github.com/humhub/cfiles/issues/127

### CODE REVIEW

File: controllers/BrowseController.php [hunk 1]
```diff
@@ -56,4 +60,37 @@ public function renderFileList($filesOrder = null, $foldersOrder = null)
         ]);
     }
 
+    public function actionLoadEntry()
+    {
+        $fileId = Yii::$app->request->get('id');
+        $fileId = strpos($fileId, 'file_') === 0 ? substr($fileId, 5) : 0;
+
+        if ($fileId && ($file = File::findOne(['id' => $fileId]))) {
```
Comment (luke-) [inline controllers/BrowseController.php:68]: Can you please add some `canView()` (Content record) or `->readable()` (ActiveQueryUser) to check access oermssions?

File: widgets/views/fileListContextMenu.php [hunk 2]
```diff
@@ -1,51 +1,21 @@
...
-</ul>
-
-<ul id="contextMenuAllPostedFiles" class="contextMenu dropdown-menu" role="menu" style="display: none">
-    <li><a tabindex="-1" href="#" data-action='download'><i class="fa fa-folder-open"></i><?= Yii::t('CfilesModule.base', 'Open'); ?></a></li>
-    <li><a tabindex="-1" href="#" data-action='show-url'><i class="fa fa-link"></i><?= Yii::t('CfilesModule.base', 'Display Url'); ?></a></li>
-</ul>
+<div data-ui-widget="stream.StreamEntry">
+    <ul class="contextMenu dropdown-menu">
+        <?php foreach ($entries as $entry) : ?>
+            <?php if($entry instanceof LegacyWallEntryControlLink) : ?>
```
Comment (luke-) [inline widgets/views/fileListContextMenu.php:12]: - Do we really require LegancyWallEntryControl Link support here?

**Review by luke- — CHANGES_REQUESTED** [review]

File: controllers/BrowseController.php (hunk 1 above)
Comment (yurabakhtin) [reply controllers/BrowseController.php]: I added the `->readable()` in commit https://github.com/humhub/cfiles/pull/131/commits/eda8f380dcd41d9f30cd753ddf260220fd13733f

File: widgets/views/fileListContextMenu.php (hunk 2 above)
Comment (yurabakhtin) [reply widgets/views/fileListContextMenu.php]: I don't find where it is used, so removed it - https://github.com/humhub/cfiles/pull/131/commits/eda8f380dcd41d9f30cd753ddf260220fd13733f

**Review by luke- — APPROVED** [review]

**Comment by luke-** [comment]
@yurabakhtin Can you please resolve the conflicts, so we can merge this?

**Comment by yurabakhtin** [comment]
@luke- 

> Can you please resolve the conflicts, so we can merge this?

Done.

**Review by luke- — APPROVED** [review]

**Pull request merged**